### PR TITLE
fix: ensure HMR requests can be streamed

### PR DESCRIPTION
**Pull request by Skn0tt**
addresses https://github.com/netlify/cli/issues/4569

### CODE REVIEW

File: src/utils/proxy.cjs [hunk 1]
```diff
@@ -379,6 +379,26 @@ const initializeProxy = async function ({ configPath, distDir, host, port, proje
     const requestURL = new URL(req.url, `http://${req.headers.host || '127.0.0.1'}`)
     const headersRules = headersForPath(headers, requestURL.pathname)
 
+    // for streamed responses, we can't do etag generation nor error templats.
```
Comment (tinfoil-knight) [inline src/utils/proxy.cjs:382]:
```suggestion
    // for streamed responses, we can't do etag generation nor error templates.
```

**Comment by Skn0tt** [comment]
tested this locally with a fresh Nuxt app. can confirm that HMR is broken on `main`, but works with this change.

**Review by eduardoboucas — APPROVED** [review]
Nice work!

**Comment by Skn0tt** [comment]
@eduardoboucas thanks for the review! could I get another ✅? I had to update a snapshot (see https://github.com/netlify/cli/pull/5194#issuecomment-1300648933).

**Review by eduardoboucas — APPROVED** [review]

**Comment by captainnurple** [comment]
Can confirm the change to `proxy.cjs` fixes hot reload for `netlify dev` in our Nuxt 2 app.

**Review by eduardoboucas — APPROVED** [review]

**Pull request merged**